kb-survey/src/app/survey/survey.component.ts

```diff
@@ -37,12 +45,38 @@ export class SurveyComponent implements OnInit {
     });
 
     this.route.queryParams.subscribe((queryParam) => {
-      const { solutionId, ...detailsOfProfile } = queryParam;
-      this.profileDetails = detailsOfProfile;
+      this.profileDetails = queryParam;
     });
-
     this.fetchSurveyDetails();
   }
+  openConfirmationDialog(confirmationParams: any): Promise<boolean> {
+    const dialogRef = this.dialog.open(DialogComponent, {
+      data: {
```

Comment:
instead of assigning each and every property we can just clone (deep copy instead of shallow copy) the `confirmationParams to data` will reduce a lot unnecessary code